A/ADOLC/build_tarballs.jl

```diff
@@ -0,0 +1,97 @@
+# Note that this script can accept some limited command-line arguments, run
+# `julia build_tarballs.jl --help` to see a usage message.
+using BinaryBuilder, Pkg
+
+# See https://discourse.julialang.org/t/binarybuilder-jl-cant-dlopen-because-of-libopenlibm-so/108486
+# This build script needs to be run with julia 1.7
+
+# See https://github.com/JuliaLang/Pkg.jl/issues/2942
+# Once this Pkg issue is resolved, this must be removed
+uuid = Base.UUID("a83860b7-747b-57cf-bf1f-3e79990d037f")
+delete!(Pkg.Types.get_last_stdlibs(v"1.6.3"), uuid)
+
+name = "ADOLC"
+version = v"0.0.24"
+
+isyggdrasil = get(ENV, "YGGDRASIL", "") == "true"
+
+# Collection of sources required to complete build
+sources = [
+    GitSource("https://github.com/coin-or/ADOL-C.git", "b7ea10536a1e452f7dd47a90ef8c8118ce3e8432"),
+    GitSource("https://github.com/TimSiebert1/libadolccxx.git", "51c23abf9ee2e1566331b440e81503ac9e7e8f5b"),
+]
+
+
+# Bash recipe for building across all platforms
+script = raw"""
+cd $WORKSPACE/srcdir
+cd ADOL-C/ && autoreconf -fi && ./configure --enable-atrig-erf --prefix=${prefix} --build=${MACHTYPE} --host=${target}
+make -j${nproc}
+make install
+mkdir ${prefix}/share/licenses/ADOLC
+install_license LICENSE
+
+cd ../libadolccxx/
+cmake -DCMAKE_INSTALL_PREFIX=${prefix}\
+      -DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN}\
+      -DCMAKE_BUILD_TYPE=Release\
+      -DADOLC_DIR=${WORKSPACE}/destdir\
+      -DJulia_PREFIX=${prefix}\
+      -B build .
+
+cd build
+make -j${nprocs}
+make install
+"""
+
+# These are the platforms we will build for by default, unless further
+# platforms are passed in on the command line
+
+platforms = [
+    Platform("x86_64", "linux"; libc = "glibc"),
+    Platform("aarch64", "macos"; ),
+    Platform("x86_64", "windows"; )
```

Comment:
Why this specific list? Do others platforms no work?

In general I would recommend to start with the full list of platforms and then use `filter!` to remove anything that is known to not work, with a comment each time as to what doesn't work.

Reply:
We had a number of problems to get cxxwrap to work. Generally, recommendations had been taken from https://discourse.julialang.org/t/binarybuilder-jl-cant-dlopen-because-of-libopenlibm-so/108486/3

Will again have a look on this. May be, also @giordano  has an update on this information.

Reply:
Looking at the Discourse thread, @giordano recommends https://github.com/JuliaPackaging/Yggdrasil/blob/9387cacc28a4eb26dd5ab0e16a78c60030e31c8a/O/openfhe_julia/build_tarballs.jl as a good example and that module does exactly what I recommended above as best practice :-)